_posts/en/newsletters/2019-11-13-newsletter.md

```diff
@@ -0,0 +1,367 @@
+  description of Sphinx that may allow a destination node to "deduce a
+  lower bound for the length of the path [back to the source node]."
+  <!-- quote from Osuntokun email -->  The fix is easy: instead of
+  initializing part of an onion packet with zero bytes, random-value
+  bytes are used instead.  Osuntokun created a [PR][lnd-onion]
+  implementing this in the onion library used by LND as well as a
+  [documentation PR][bolts #697] for the BOLTs repository.  Other
+  implementations have also adopted the same change (see the
+  [C-Lightning commit][news72 cl onion] below).
+
+- **LN up-front payments:** the current LN protocol returns all money to
+  the spender if a payment attempt fails or is rejected by the receiver,
+  so routing nodes only receive income if the payment attempt succeeds.
+  However, some new applications are using this costless failure
+  mechanism to send data over LN without paying for the bandwidth they
+  use.  LN designers expected this to happen and have previously spent
+  time thinking about how they could add up-front fees to the
+  network---fees that would be paid to routing nodes whether or not a
+  payment attempt succeeded.
+
+    This week, Rusty Russell started a [thread][russell up-front] on the
+    Lightning-Dev mailing list to discuss proposals for up-front fees.
+    Russell proposed a mechanism that combines fees and hashcash-style
+    proof-of-work to try to prevent nodes from using using the extra
+    up-front payment information to guess the length of the route.
+    Anthony Towns proposed a partial [alternative][towns up-front]
+    focusing on managing payment amounts using a refund mechanism.
+
+    Joost Jager suggested that up-front payments should only be required
+    as a last resort because even small additional fees could make
+    micropayments uneconomic.  He suggested that it should be possible
+    to address bandwidth-wasting network activity using rate limits
+    based on node reputation and, further, that research into up-front
+    payments should focus on first solving liquidity abuse---where an
+    attacker ties up someone's in-channel funds for period of time---as
+    the solution to that problem may also prevent the abuse of routing
+    node bandwidth.
+
+    Ultimately, no conclusions were reached and discussion about the
+    topic remains ongoing as of this writing.
+
+- **Proposed BOLT for LN offers:** Rusty Russell [posted][bolt offers]
+  draft text for a new BOLT that would allow users to submit offers and
+  receive invoices over the LN routing protocol.  For example, Alice
+  could subscribe to a service provided by Bob where each month she
+  would submit an offer to pay Bob, Bob would reply with an invoice,
+  Alice would pay the invoice, and Bob would provide the service.
+
+    Early feedback on the proposal suggested that it might want to
+    incorporate by reference an established language for
```

Comment:
'it might want to incorporate by reference..' is a little difficult to parse. I think either 'incorporate' or 'reference' are fine, but using both is too much.

Reply:
oh interesting. Thanks!